zapisy/apps/enrollment/timetable/assets/components/filters/MultiSelectFilter.vue

```diff
@@ -0,0 +1,316 @@
+  },
+  unmounted() {
+    window.removeEventListener("resize", this.updateDropdownWidth);
+  },
+  methods: {
+    ...mapMutations("filters", ["registerFilter"]),
+    clearFilter() {
+      this.selected = [];
+    },
+    clearAll() {
+      this.selected = [];
+    },
+    updateDropdownWidth() {
+      const multiselectInputs =
+        document.querySelectorAll<HTMLElement>(".multiselect");
+
+      Array.from(multiselectInputs).forEach((multiselectInput) => {
+        const dropdown = multiselectInput.querySelectorAll<HTMLElement>(
+          ".multiselect__content-wrapper"
+        )[0];
+
+        if (dropdown) {
+          dropdown.style.width = `${multiselectInput.offsetWidth}px`;
+        }
+      });
+    },
+  },
+  computed: {
+    selectionDescription(): string {
+      if (this.selected.length === 0) {
+        return this.placeholder;
+      }
+      return this.options
+        .filter((opt) => this.selected.includes(opt))
+        .map(({ label }: { label: string }) => label)
+        .join(", ");
+    },
+  },
+  watch: {
+    selected: function () {
+      const selectedIds = this.selected.map(
+        (selectedFilter: Option) => selectedFilter.value
+      );
+
+      const url = new URL(window.location.href);
+      if (isEmpty(selectedIds)) {
+        url.searchParams.delete(this.property);
+      } else {
+        url.searchParams.set(this.property, selectedIds.join(","));
+      }
+      window.history.replaceState(null, "", url.toString());
+
+      this.registerFilter({
+        k: this.filterKey,
+        f: new ExactFilter(selectedIds, this.property),
+      });
+    },
+  },
+});
+</script>
+
+<template>
+  <div class="mb-2">
+    <multiselect
+      v-model="selected"
+      :options="options"
+      :show-labels="showLabels"
+      :multiple="true"
+      :close-on-select="false"
+      :track-by="trackBy"
+      :label="propAsLabel"
+      :placeholder="selectionDescription"
+    >
+      <template slot="option" slot-scope="props">
+        <div class="option-row">
+          <div class="custom-control custom-checkbox">
+            <input
+              type="checkbox"
+              class="custom-control-input"
+              :checked="selected.includes(props.option)"
+            />
+            <label class="custom-control-label" :for="filterKey"></label>
+          </div>
+          {{ props.option.label }}
+        </div>
+      </template>
+      <template slot="selection" slot-scope="{ values, isOpen }">
+        <span class="multiselect__single" v-if="values.length" v-show="!isOpen">
+          {{ selectionDescription }}
+        </span>
+      </template>
+      <template slot="clear">
+        <div
+          v-show="selected.length"
+          class="multiselect__clear"
+          @mousedown.prevent.stop="clearAll"
+        >
+          ×
+        </div>
+      </template>
+    </multiselect>
+  </div>
+</template>
+
+<style src="vue-multiselect/dist/vue-multiselect.min.css"></style>
+
+<style>
```

Comment:
Spodziewałbym się czegoś z grubsza takiego
```suggestion
<style lang="scss" scoped>
@import "~vue-multiselect/dist/vue-multiselect.min.css";
```
ale proszę samemu sprawdzić/dowiedzieć się, czy ta propozycja (uwzględniając wszystkie jej elementy) ma sens, ja bardziej zgaduję jej formę na podstawie tego, jak wyglądają inne pliki w projekcie. W szczególności nigdzie nie używamy dyrektywy `@import` w plikach `.vue`, więc gdyby z jakiegoś kryptycznego powodu nie działała ona tutaj, to będzie należało wywalić to do osobnego pliku `.scss` (ale poza tą ewentualnością nie widzę w tym momencie takiej potrzeby, optycznie nie wygląda ta sekcja jakoś strasznie).

Reply:
Rzeczywiście, wygląda to lepiej i działa.
Problem jest jedynie z atrybutem "scoped": style wypisane w takim bloku nie nakładają się na te zaimportowane. Mogę więc albo usunąć ten atrybut, albo, jeśli miałoby to w czymś pomóc, zrobić dwa bloki: jeden "scoped" z importem, drugi, pozbawiony go, z całą resztą.

Reply:
Ciekawe, z czego to się bierze; zgaduję, że jest ortogonalne do reszty proponowanej przeze mnie zmiany – czy jeśli będą dwa osobne elementy `<style>` (w "starej" bądź "nowej" formie, to już chyba nie powinno mieć znaczenia), to które z czterech możliwości nadania im tego atrybutu działają, a które nie?

Dwa bloki to oczywiście żadna herezja, jest na to nawet przykład w https://vuejs.org/api/sfc-css-features.html ale dobrze byłoby to robić wtedy, kiedy chcemy, a nie wtedy, kiedy jesteśmy do tego zmuszeni.

Reply:
Zgaduję, że problem wynika wprost z semantyki atrybutu `scoped`: jeśli dobrze rozumiem, to zapewnia on, że style pod nim zadeklarowane obowiązywać będą tylko w komponencie, który ten styl zawiera.
W tym przypadku: komponentem zawierającym styl jest nasz `MultiSelectFilter`, ale chcemy, aby te style dotyczyły zagnieżdżonego w nim komponentu `multiselect`.
Dlaczego deklaracja `@import` działa ze `scoped` nie rozumiem, ale być może jest jakieś znane silnikowi mapowanie pomiędzy stylami, które importujemy, a komponentem `multiselect`, dzięki któremu Vue wie, że właśnie w nim powinien te style zastosować.